File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Azerbaijan.m
//
//  UIImage+Azerbaijan.m
//  CountriesFlags
//

#import "Countries/UIImage+Azerbaijan.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (Azerbaijan)

+ (UIImage * _Nullable)flagAzerbaijanWithSize:(CGSize)size {
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorBlue = [UIColor colorWithRed:0.0f
                                         green:181.0f/255.0f
                                          blue:226.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:239.0f/255.0f
                                        green:54.0f/255.0f
                                         blue:61.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorGreen = [UIColor colorWithRed:80.0f/255.0f
                                          green:158.0f/255.0f
                                           blue:47.0f/255.0f
                                          alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    [colorWhite setFill];
    [[UIBezierPath bezierPathWithArcCenter:CGPointMake(size.width / 1920.0f * 912.0f, size.height / 2.0f)
                                    radius:size.height / 10.0f * 3.0f / 2.0f
                                startAngle:0.0f
                                  endAngle:M_PI * 2.0f
                                 clockwise:YES] fill];

    [colorRed setFill];
    [[UIBezierPath bezierPathWithArcCenter:CGPointMake(size.width / 1920.0f * 912.0f + size.height / 30.0f, size.height / 2.0f)
                                    radius:size.height / 4.0f / 2.0f
                                startAngle:0.0f
                                  endAngle:M_PI * 2.0f
                                 clockwise:YES] fill];
    
    [colorWhite setFill];
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 1920.0f * 1074.0f, size.height / 2.0f)
                        radius:size.height / 12.0f
                          type:FlagStarTypeEightPointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Azerbaijan.swift
//
//  UIImage+Azerbaijan.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Azerbaijan "wikipedia"
     
     Generates the flag of Azerbaijan.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Azerbaijan.
     */
    internal static func flagAzerbaijan(with size: CGSize) -> UIImage? {
        let colorWhite = UIColor.white
        let colorBlue = UIColor(red: 0.0,
                                green: 181.0 / 255.0,
                                blue: 226.0 / 255.0,
                                alpha: 1.0)
        let colorRed = UIColor(red: 239.0 / 255.0,
                               green: 54.0 / 255.0,
                               blue: 61.0 / 255.0,
                               alpha: 1.0)
        let colorGreen = UIColor(red: 80.0 / 255.0,
                                 green: 158.0 / 255.0,
                                 blue: 47.0 / 255.0,
                                 alpha: 1.0)
        let lineSize = CGSize(width: size.width, height: size.height / 3.0)

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorRed.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0), size: lineSize))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0 * 2.0), size: lineSize))

        colorWhite.setFill()
        UIBezierPath(arcCenter: CGPoint(x: size.width / 1920.0 * 912.0, y: size.height / 2.0),
                     radius: size.height / 10.0 * 1.5,
                     startAngle: 0.0,
                     endAngle: .pi * 2.0,
                     clockwise: true).fill()

        colorRed.setFill()
        UIBezierPath(arcCenter: CGPoint(x: size.width / 1920.0 * 912.0 + size.height / 30.0, y: size.height / 2.0),
                     radius: size.height / 8.0,
                     startAngle: 0.0,
                     endAngle: .pi * 2.0,
                     clockwise: true).fill()

        colorWhite.setFill()
        UIBezierPath.star(at: CGPoint(x: size.width / 1920.0 * 1074.0, y: size.height / 2.0),
                          radius: size.height / 12.0,
                          type: .eightPointed).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
